contracts/HtsSystemContractJson.sol

```diff
@@ -463,6 +463,17 @@ contract HtsSystemContractJson is HtsSystemContract {
         return slot;
     }
 
+    function __nftInfoSlot(uint32 serialId) internal override virtual returns (bytes32) {
+        bytes32 slot = super.__nftInfoSlot(serialId);
+        if (_shouldFetch(slot)) {
+            string memory metadata = mirrorNode().getNftMetadata(address(this), serialId);
+            string memory createdTimestamp = mirrorNode().getNftCreatedTimestamp(address(this), serialId);
+            int64 creationTime = int64(vm.parseInt(vm.split(createdTimestamp, ".")[0]));
+            storeBytes(address(this), uint256(slot), abi.encode(creationTime, bytes(metadata)));
```

Comment:
we should also set the flag in `_scratchAddr()` to prevent re-fetching (see `_setValue` function).

Also, please modify `_tokenUriSlot` because it doesn't update the `_scratchAddr()` either.